### PR TITLE
User activity paging

### PR DESCRIPTION
**Pull request by ajacksified**
:eyeglasses: @curioussavage or @schwers 

https://reddit.atlassian.net/browse/MOBILEWEB-342

Adds paging for user profile activity and user saved/hidden pages. Also fixes a bug with index paging.

### CODE REVIEW

**Comment by ajacksified** [comment]
Depends on https://github.com/reddit/snoode/pull/67

**Comment by ajacksified** [comment]
We may want to pull paging into the listing component as part of the redesign. It would also be nice if the listingcontainer watched `compact` toggles itself, rather than every page implementing it.

File: src/views/components/ListingContainer.jsx [hunk 1]
```diff
@@ -31,8 +33,26 @@ class ListingContainer extends BaseComponent {
 
   render() {
     const props = this.props;
+    const { listings, ctx, shouldPage, pagingPrefix, prevUrl, nextUrl } = props;
```
Comment (schwers) [inline src/views/components/ListingContainer.jsx]: Can you add the new paging props to the `static propTypes`

File: src/views/components/ListingPaginationButtons.jsx [hunk 2]
```diff
@@ -11,9 +12,57 @@ class ListingPaginationButtons extends BaseComponent {
...
+    return `${pagingPrefix || ''}?${querystring.stringify(nextQuery)}`;
+  }
+
   render() {
     const compact = this.state.compact;
-    const { prevUrl, nextUrl } = this.props;
+
+    // Allow overriding for special cases, like search. Otherwise, fall back to
+    // the default logic for building next/previous urls.
+    let { prevUrl, nextUrl } = this.props;
```
Comment (schwers) [inline src/views/components/ListingPaginationButtons.jsx]: Same as `ListingContainer`, can you update propTypes for the documentation / warning benefits

File: src/views/components/ListingPaginationButtons.jsx [hunk 3]
```diff
@@ -11,9 +12,57 @@ class ListingPaginationButtons extends BaseComponent {
...
+
+    const nextQuery = {
+      ...props.ctx.query,
+      count: 25,
+      page: page + 1,
+      after: lastId,
+      before: undefined,
+    };
+
+    return `${pagingPrefix || ''}?${querystring.stringify(nextQuery)}`;
```
Comment (schwers) [inline src/views/components/ListingPaginationButtons.jsx]: :no_bell: TIL you can also do `const { pagingPrefix = '' } = props;` to declare the default for a var when destructuring. It does add an extra var assignment in the transpiled js. What do you think?

File: src/views/pages/userActivity.jsx [hunk 4]
```diff
@@ -39,8 +39,8 @@ class UserActivityPage extends BasePage {
         />
 
         <ListingContainer
-          compact={ props.compact }
```
Comment (schwers) [inline src/views/pages/userActivity.jsx]: Removing this breaks the userActivity page respecting compact

**Comment by schwers** [comment]
:nail_care: Mostly small nits

**Comment by ajacksified** [comment]
:haircut:

File: src/views/pages/userActivity.jsx (hunk 4 above)
Comment (ajacksified) [inline src/views/pages/userActivity.jsx]: [added]

File: src/views/components/ListingPaginationButtons.jsx (hunk 3 above)
Comment (ajacksified) [inline src/views/components/ListingPaginationButtons.jsx]: Neat, looks cleaner.

File: src/views/components/ListingPaginationButtons.jsx (hunk 2 above)
Comment (ajacksified) [inline src/views/components/ListingPaginationButtons.jsx]: Updated.

File: src/views/components/ListingPaginationButtons.jsx (hunk 3 above)
Comment (curioussavage) [inline src/views/components/ListingPaginationButtons.jsx]: I like this too.

File: src/views/components/ListingPaginationButtons.jsx [hunk 5]
```diff
@@ -11,9 +12,61 @@ class ListingPaginationButtons extends BaseComponent {
...
+      pagingPrefix = '',
+      listings,
+      ctx,
+    } = this.props;
+
+    const firstId = listings[0].name;
+    const page = parseInt(ctx.query.page || 0);
+
+    if (page > 0) {
+      let prevQuery = {
```
Comment (curioussavage) [inline src/views/components/ListingPaginationButtons.jsx]: const?

**Comment by curioussavage** [comment]
:fish: code looks good to me. just that one const.

**Comment by schwers** [comment]
:fish:

**Pull request merged**